clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -215,13 +182,14 @@ def _is_node_safe_to_kill(self, node_ip: str) -> bool:
         return True
 
     def _get_nodes_by_ip(self) -> Mapping[str, KubernetesNode]:
-        pool_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
-        pool_nodes = self._core_api.list_node().items
+        node_label_selector = self.pool_config.read_string("node_label_key", default="clusterman.com/pool")
+        label_selector = f"{node_label_selector}={self.pool}"
+        pool_nodes = self._core_api.list_node(label_selector=label_selector).items
 
         return {
             get_node_ip(node): node
             for node in pool_nodes
-            if not self.pool or node.metadata.labels.get(pool_label_selector, None) == self.pool
+            if not self.pool or node.metadata.labels.get(node_label_selector, None) == self.pool
```

Comment:
Do you really need this?

Reply:
Because you already added label selector to your apiserver query

Reply:
This fixed.